src/Compilers/CSharp/Portable/Binder/AwaitableInfo.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    /// <summary>
+    /// Internal structure containing all semantic information about an await expression.
+    /// </summary>
+    internal class AwaitableInfo : IEquatable<AwaitableInfo>
+    {
+        public MethodSymbol GetAwaiter { get; }
+
+        public PropertySymbol IsCompleted { get; }
+
+        public MethodSymbol GetResult { get; }
+
+        public TypeSymbol Type { get; }
```

Comment:
>Type [](start = 26, length = 4)

What is this type? Consider adding a comment. Also, it feels like we are duplicating information that can be inferred from other members. I would try to get rid of it.